src/cmdline.rs

```diff
@@ -201,11 +201,22 @@ pub fn parse() -> Result<Command> {
     } else if let Some(mut args) = cmd {
         if let Some(exe) = args.next() {
             let cmdline = args.map(|s| s.to_owned()).collect::<Vec<_>>();
+            let mut env_vars = env::vars_os().collect::<Vec<_>>();
+
+            // If we're running under rr, avoid the `LD_PRELOAD` bits, as it will
+            // almost surely do the wrong thing, as the compiler gets executed
+            // in a different process tree.
```

Comment:
Wait, this happens even in the local case, not just the dist case?  So where you'd expect `rr -> sccache -> compiler`, you actually get `rr -> sccache` and `rr -> compiler` ?  Who's separating things out into a different process tree?

Reply:
rr -> sccache -> sccache daemon -> compiler?

Reply:
All the builds go through the daemon, afaict, which is what decides whether to remote again to a remote builder or not... My understanding here could be wrong, but I can trivially reproduce it even off-line with something like: `$ rr sccache /usr/bin/gcc -c t.c -o t.o`, where `t.c` is just `int main() { return 0; }`

Reply:
Ah, right.  OK.